### PR TITLE
fix: normalize volatile_state.location to loc-* entity IDs

### LINKED ISSUE #322: fix: normalize volatile_state.location to loc-* IDs
## Summary

Entity detail extraction populates `volatile_state.location` with **free-text descriptions** instead of `loc-*` IDs from the location catalog. This makes spatial queries (scene graph, co-location filtering) unreliable or non-functional.

## Problem

After 344 turns of extraction (293 entities, 92 locations), **zero** entities have a `loc-*` ID in their `volatile_state.location` field. Instead, the LLM writes descriptions like:

- `"At the healer's location, undergoing care"` (should be `loc-healers-hut` or similar)
- `"Longhouse"` (should be `loc-longhouse`)
- `"Boundary area / Patrol route"` (should be `loc-boundary` or similar)
- `"With the player and the man (working on disrupting the pattern...)"` (not a location at all)

The same physical location appears under 168 unique string variants, preventing:

1. **Scene graph builder** (`tools/build_scene_graph.py`): filters on `loc-.startswith("loc-")` — produces empty `location_index` with current data
2. **Co-location matching** (`tools/catalog_merger.py`): uses exact string equality — catches only ~14% of entities in 13 groups, misses obvious equivalences like `"Longhouse"` vs `"near the longhouse"`
3. **Spatial scoping for discovery prompts** (#310 Phase 3.2): cannot filter entities by location proximity without normalized IDs

### 1. Template guidance (entity-detail.md)

Update the entity detail extraction template to instruct the LLM:
- Use the `loc-*` ID from the known-entity list when the entity is at a cataloged location
- Fall back to a short place name only if no matching location exists

### 2. Post-extraction normalization

Add a programmatic normalization step after entity detail extraction that:
- Checks if `volatile_state.location` is a free-text string (not a `loc-*` ID)
- Fuzzy-matches against known location names and aliases from the catalog
- Replaces with the `loc-*` ID if a confident match is found
- Logs replacements for auditability

This is similar to the existing `_normalize_entity_id()` pattern that strips turn-tag suffixes.

### 3. Downstream benefits

Once locations are normalized to `loc-*` IDs:
- `build_scene_graph.py` produces a populated `location_index`
- Co-location matching in `catalog_merger.py` works reliably
- Spatial scoping (#310 Phase 3.2) becomes viable
- Entity refresh can prioritize entities at the player's current location

## Acceptance Criteria

- [ ] Entity detail template instructs LLM to use `loc-*` IDs
- [ ] Post-extraction normalization maps free-text to `loc-*` IDs when confident match exists
- [ ] Normalization logs replacements to stderr
- [ ] Unit tests for the normalization function
- [ ] `build_scene_graph.py` produces non-empty `location_index` with normalized data
- [ ] Existing tests still pass

## Related

- Part of #310 (Phase 3.2 prerequisite)
- Uses location catalog from `framework/catalogs/locations/`
- Scene graph schema: `schemas/scene-graph.schema.json`

### PR DESCRIPTION
**Pull request by daviburg**
## Summary

Normalizes `volatile_state.location` from free-text descriptions to `loc-*` entity IDs, enabling reliable spatial queries.

## Problem

After 344 turns of extraction (293 entities, 92 locations), **zero** entities had a `loc-*` ID in their `volatile_state.location`. Instead, the LLM wrote free-text like `"At the healer's location, undergoing care"` or `"Longhouse"`. This made:

- `build_scene_graph.py` produce empty `location_index` (filters on `loc-*` prefix)
- Co-location matching catch only ~14% of entities via exact string equality
- Spatial scoping for discovery prompts (#310 Phase 3.2) non-viable

### Post-extraction normalization

`_normalize_entity_location(entity_data, catalogs)` in `semantic_extraction.py`:
- Checks if `volatile_state.location` matches a known location's name or alias (case-insensitive)
- Replaces with the `loc-*` ID on match
- Logs replacements to stderr as `NORMALIZE_LOC:`
- Called at all 6 `merge_entity()` sites (parallel detail, PC ×2, non-parallel detail, stub backfill, entity refresh)

### Template guidance

Updated `templates/extraction/entity-detail.md`:
- Location field now instructs: "Use the `loc-*` entity ID when the entity is at a location that exists in the known-entity list"
- Example updated to show `loc-council-fire` instead of `"council fire"`

### Tests

12 new tests in `tests/test_location_normalization.py` covering:
- Exact name match, case-insensitive, alias match
- No match (preserves original), already normalized (no-op)
- Edge cases: no volatile_state, no location, empty string, whitespace
- Alias as string vs list, empty catalogs, duplicate location names

### Documentation

Updated `docs/architecture.md` with Location ID normalization entry.

Fixes #322

### CODE REVIEW

File: tools/semantic_extraction.py [hunk 1]
```diff
@@ -293,6 +293,48 @@ def _load_schema(name: str) -> dict | None:
...
+    if not loc_value or not isinstance(loc_value, str):
+        return
+    # Already a loc-* ID
+    if loc_value.startswith("loc-"):
+        return
+
+    loc_lower = loc_value.lower().strip()
+    if not loc_lower:
+        return
+
```
[the remark this answers is not in the document]
Comment (daviburg) [reply tools/semantic_extraction.py]: Fixed in e7bbf68. Now strips whitespace and checks for loc- prefix case-insensitively before short-circuiting. Uppercase/mixed-case variants are canonicalized to lowercase.

File: tools/semantic_extraction.py [hunk 2]
```diff
@@ -293,6 +293,48 @@ def _load_schema(name: str) -> dict | None:
...
+    # Build name→ID lookup from location catalog
+    locations = catalogs.get("locations.json", [])
+    for loc_entity in locations:
+        loc_id = loc_entity.get("id", "")
+        # Check name
+        if loc_entity.get("name", "").lower().strip() == loc_lower:
+            vol["location"] = loc_id
+            print(f"  NORMALIZE_LOC: {loc_value!r} -> {loc_id}", file=sys.stderr)
+            return
```
[the remark this answers is not in the document]
Comment (daviburg) [reply tools/semantic_extraction.py]: Fixed in e7bbf68. Now collects all matches and skips normalization (preserves original free-text) when multiple locations share the same name/alias. Logs a warning to stderr.

File: tests/test_location_normalization.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        entity = {"volatile_state": {"location": "the dark forest"}}
+        _normalize_entity_location(entity, catalogs)
+        assert entity["volatile_state"]["location"] == "the dark forest"
+
+    def test_already_loc_id_unchanged(self):
+        catalogs = _make_catalogs([_make_loc("loc-longhouse", "Longhouse")])
+        entity = {"volatile_state": {"location": "loc-longhouse"}}
+        _normalize_entity_location(entity, catalogs)
+        assert entity["volatile_state"]["location"] == "loc-longhouse"
+
```
[the remark this answers is not in the document]
Comment (daviburg) [reply tests/test_location_normalization.py:56]: Fixed in e7bbf68. Added 3 test cases: leading/trailing whitespace, uppercase prefix, and mixed case+whitespace (' LOC-Longhouse ').
